cypress/support/stackable/helpers/index.js

```diff
@@ -437,8 +437,81 @@ export const assertContainerLink = ( selector, options = {}, assertOptions = {}
 				cy.adjust( 'Link / URL', 'https://www.google.com/' ).assertHtmlAttribute( `${ selector } > a`, 'href', 'https://www.google.com/', assertOptions )
 				cy.adjust( 'Link Title', 'Sample Link Title' ).assertHtmlAttribute( `${ selector } > a`, 'title', 'Sample Link Title', assertOptions )
 			}
-			cy.adjust( 'Open link in new tab', true ).assertHtmlAttribute( `${ selector } > a`, 'rel', 'noopener noreferrer', assertOptions )
-			cy.adjust( 'Nofollow link', true ).assertHtmlAttribute( `${ selector } > a`, 'rel', 'noopener noreferrer nofollow', assertOptions )
+			cy.adjust( 'Open link in new tab', true ).assertHtmlAttribute( `${ selector } > a`, 'rel', /noopener noreferrer/, Object.assign( assertOptions, { assertBackend: false } ) )
+			cy.adjust( 'Nofollow link', true ).assertHtmlAttribute( `${ selector } > a`, 'rel', /nofollow/, Object.assign( assertOptions, { assertBackend: false } ) )
+			cy.adjust( 'Sponsored', true ).assertHtmlAttribute( `${ selector } > a`, 'rel', /sponsored/, Object.assign( assertOptions, { assertBackend: false } ) )
+			cy.adjust( 'UGC', true ).assertHtmlAttribute( `${ selector } > a`, 'rel', /ugc/, Object.assign( assertOptions, { assertBackend: false } ) )
+		} )
+	} )
+}
+
+/**
+ * Helper function for asserting the URL popover of the buttons.
+ *
+ * @param {string} blockName
+ * @param {string | number | Object} blockSelector
+ * @param {Object} selectors
+ * @param {Object} options
+ * @param {Object} assertOptions
+ */
+export const assertUrlPopover = ( blockName, blockSelector, selectors = {}, options = {}, assertOptions = {} ) => {
+	const {
+		viewport = 'Desktop',
+	} = options
+	const {
+		editorSelector = '',
+		frontendSelector = '',
+	} = selectors
+	const desktopOnly = callback => viewport === 'Desktop' && callback()
+
+	desktopOnly( () => {
+		cy.get( '.is-selected' ).then( $block => {
+			// Count the number of buttons
+			const count = $block.find( editorSelector.replace( '%s', '' ) ).length
+			if ( count === 0 ) {
+				return
+			}
+
+			const parentSelector = '.components-popover__content .block-editor-link-control'
+			const supportedDelimiter = [ ' ' ]
+
+			range( 1, count + 1 ).forEach( number => {
+				cy.get( '.is-selected' ).find( editorSelector.replace( '%s', number ) ).click( { force: true } )
+
+				/**
+				 * This will not work on dynamic blocks since we need to do some extra steps
+				 * before adjusting the options again.
+				 */
```

Comment:
```suggestion
				/**
				 * TODO: This will not work on dynamic blocks since we need to do some extra steps
				 * before adjusting the options again.
				 */
```